internal/cssc/cssc.go

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package cssc
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/Azure/acr-cli/internal/tag"
+
+	"github.com/Azure/acr-cli/cmd/api"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+
+	oras "oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content"
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+	"oras.land/oras-go/v2/registry/remote/retry"
+)
+
+// Filter struct to hold the filter policy
+type Filter struct {
+	Version      string       `json:"version"`
+	Repositories []Repository `json:"repositories"`
+}
+
+type Repository struct {
+	Repository string   `json:"repository"`
+	Tags       []string `json:"tags"`
+	Enabled    *bool    `json:"enabled"`
+}
+
+// Repository struct to hold the filtered repository, tag and patch tag if any
+type FilteredRepository struct {
+	Repository string
+	Tag        string
+	PatchTag   string
+}
+
+// Reads the filter policy from the specified repository and tag and returns the Filter struct
+func GetFilterFromFilterPolicy(ctx context.Context, filterPolicy string, loginURL string, username string, password string) (Filter, error) {
+	if !strings.Contains(filterPolicy, ":") {
+		return Filter{}, errors.New("filter-policy should be in the format repo:tag")
+	}
+	repoTag := strings.Split(filterPolicy, ":")
+	filterRepoName := repoTag[0]
+	filterRepoTagName := repoTag[1]
+
+	// Connect to the remote repository
+	repo, err := remote.NewRepository(fmt.Sprintf("%s/%s", loginURL, filterRepoName))
+	if err != nil {
+		panic(err)
```

Comment:
You should not be panicking here since the command itself can propagate and possibly return the error to the user. Instead wrap the error and return it up.

Reply:
Addressed this in the latest commit.